py/grammar.h

```diff
@@ -51,7 +51,9 @@ DEF_RULE_NC(file_input_3, or(2), tok(NEWLINE), rule(stmt))
 DEF_RULE_NC(eval_input, and_ident(2), rule(testlist), opt_rule(eval_input_2))
 DEF_RULE_NC(eval_input_2, and(1), tok(NEWLINE))
 
-// decorator: '@' dotted_name [ '(' [arglist] ')' ] NEWLINE
+// decorator: '@' decorator_arg NEWLINE
+// decorator_arg: built_in_decorator | namedexpr_test
+// built_in_decorator: 'micropython' '.' NAME
```

Comment:
I think it might be cleaner/simpler to define the grammar like this:
```
decorator: '@' namedexpr_test NEWLINE
```
and then handle detection and compilation of the builtin decorator in `py/compile.c` (that way it can potentially support more complex builtin decorators, eg `@micropython.bytecode(opt=2)`).